yt/visualization/_handlers.py

```diff
@@ -0,0 +1,362 @@
+            else:
+                norm_type = LogNorm
+
+        if norm_type is SymLogNorm:
+            # if cblinthresh is not specified, try to come up with a reasonable default
+            min_abs_val, max_abs_val = np.sort(
+                np.abs((self.to_float(np.nanmin(data)), self.to_float(np.nanmax(data))))
+            )
+            if self.linthresh is not None:
+                linthresh = self.to_float(self.linthresh)
+            elif min_abs_val > 0:
+                linthresh = min_abs_val
+            else:
+                linthresh = max_abs_val / 1000
+            kw.setdefault("linthresh", linthresh)
+            if MPL_VERSION >= Version("3.2"):
+                # note that this creates an inconsistency between mpl versions
+                # since the default value previous to mpl 3.4.0 is np.e
+                # but it is only exposed since 3.2.0
+                kw.setdefault("base", 10)
+
+        return norm_type(*args, **kw)
+
+
+class ColorbarHandler:
+    __slots__ = ("_draw_cbar", "_draw_minorticks", "_cmap", "_background_color")
+
+    def __init__(
+        self,
+        *,
+        draw_cbar: bool = True,
+        draw_minorticks: bool = True,
+        cmap: Optional[Union[Colormap, str]] = None,
+        background_color: Optional[str] = None,
+    ):
+        self._draw_cbar = draw_cbar
+        self._draw_minorticks = draw_minorticks
+        self._cmap: Optional[Colormap] = None
+        self.cmap = cmap
+        self._background_color = background_color
+
+    @property
+    def draw_cbar(self) -> bool:
+        return self._draw_cbar
+
+    @draw_cbar.setter
+    def draw_cbar(self, newval) -> None:
+        if not isinstance(newval, bool):
+            raise TypeError(
+                f"Excpected a boolean, got {newval} with type {type(newval)}"
+            )
+        self._draw_cbar = newval
+
+    @property
+    def draw_minorticks(self) -> bool:
+        return self._draw_minorticks
+
+    @draw_minorticks.setter
+    def draw_minorticks(self, newval) -> None:
+        if not isinstance(newval, bool):
+            raise TypeError(
+                f"Excpected a boolean, got {newval} with type {type(newval)}"
+            )
+        self._draw_minoticks = newval
+
+    @property
+    def cmap(self) -> Colormap:
+        return self._cmap or get_cmap(ytcfg.get("yt", "default_colormap"))
+
+    @cmap.setter
+    def cmap(self, newval) -> None:
+        if isinstance(newval, Colormap) or newval is None:
+            self._cmap = newval
+        elif isinstance(newval, str):
+            self._cmap = get_cmap(newval)
+        elif is_sequence(newval):
+            # tuple colormaps are from palettable (or brewer2mpl)
+            self._cmap = get_brewer_cmap(newval)
+        else:
+            raise TypeError(
+                "Expected a colormap object or name, "
+                f"got {newval} with type {type(newval)}"
+            )
+
+    @property
+    def background_color(self) -> str:
```

Comment:
I couldn't find where that was documented (I looked up matplotlib's source and docs)
https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.set_facecolor.html?highlight=set_facecolor#matplotlib.axes.Axes.set_facecolor
We also don't have a single example of this use case anywhere (no test and no docs).
I agree this `ImagePlotContainer.set_background_color()`'s docstring is out of sync with type hints, I just can't tell which way it should be corrected

Reply:
Ya, hmm... I guess matplotlib doesn't formally document it in the source. the linked demo does though https://matplotlib.org/stable/gallery/color/color_demo.html#color-demo

Reply:
Maybe don't worry about it here but we could open an issue to look into it and maybe add some tests on the yt side using tuples?

Reply:
Note, I don't think yt needs to support all those color specifications described in the demo. IMO, having tuples + strings should be sufficient.

Reply:
I think I'm just going to type this as `Any` and explain why in a comment, because I see no gain in trying to match the exact input types supported by `matplotlib.axes.Axes.set_faceolor`: it's not just strings and 4-tuples, any 3 or 4 element float sequence should work as well, and we could also express the value constraints in the type...

Reply:
@chrishavlin